VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -413,142 +413,157 @@ public void testUnitFragmentCreation() {
+        }
 
         CourseComponent nonMultiDeviceModel = Mockito.mock(CourseComponent.class);
         when(nonMultiDeviceModel.isMultiDevice()).thenReturn(false);
         unitList.add(nonMultiDeviceModel);
+        classesList.add(CourseUnitMobileNotSupportedFragment.class);
 
         HtmlBlockModel htmlModel = Mockito.mock(HtmlBlockModel.class);
         when(htmlModel.isMultiDevice()).thenReturn(true);
         when(htmlModel.getType()).thenReturn(BlockType.HTML);
         unitList.add(htmlModel);
+        classesList.add(CourseUnitWebViewFragment.class);
 
         CourseComponent unknownModel = Mockito.mock(CourseComponent.class);
         when(unknownModel.isMultiDevice()).thenReturn(true);
         when(unknownModel.getType()).thenReturn(BlockType.COURSE);
         unitList.add(unknownModel);
+        classesList.add(CourseUnitEmptyFragment.class);
 
         CourseComponent problemModel = Mockito.mock(CourseComponent.class);
         when(problemModel.isMultiDevice()).thenReturn(true);
         when(problemModel.getType()).thenReturn(BlockType.PROBLEM);
         unitList.add(problemModel);
+        classesList.add(CourseUnitMobileNotSupportedFragment.class);
 
         CourseComponent othersModel = Mockito.mock(CourseComponent.class);
         when(othersModel.isMultiDevice()).thenReturn(true);
         when(othersModel.getType()).thenReturn(BlockType.OTHERS);
         unitList.add(othersModel);
+        classesList.add(CourseUnitMobileNotSupportedFragment.class);
 
         CourseUnitPagerAdapter adapter = new CourseUnitPagerAdapter(fragmentManager, config,
                 unitList, courseData, hasComponent);
 
-        assertTrue(adapter.getItem(0) instanceof CourseUnitVideoFragment);
-        assertTrue(adapter.getItem(1) instanceof CourseUnitOnlyOnYoutubeFragment);
-        if (config.isDiscussionsEnabled()) {
-            assertTrue(adapter.getItem(2) instanceof CourseUnitDiscussionFragment);
-        } else {
-            assertTrue(adapter.getItem(2) instanceof CourseUnitMobileNotSupportedFragment);
+        for (int size = unitList.size(), i = 0; i < size; i++) {
+            assertThat(adapter.getItem(i)).isInstanceOf(classesList.get(i));
         }
-        assertTrue(adapter.getItem(3) instanceof CourseUnitMobileNotSupportedFragment);
-        assertTrue(adapter.getItem(4) instanceof CourseUnitWebViewFragment);
-        assertTrue(adapter.getItem(5) instanceof CourseUnitEmptyFragment);
-        assertTrue(adapter.getItem(6) instanceof CourseUnitMobileNotSupportedFragment);
-        assertTrue(adapter.getItem(7) instanceof CourseUnitMobileNotSupportedFragment);
     }
 
+    /**
+     * Tests creation of various fragments in the {@link CourseUnitNavigationActivity}'s
+     * ViewPager, by supplying its {@link CourseUnitPagerAdapter} with a single
+     * {@link CourseComponent} model at a time using an input parameter and verifying the
+     * creation by providing an expected output parameter.
```

Comment:
"using an input parameter and verifying the creation by providing an expected output parameter." could be simplified to just "via parameterization".